src/DynamoUtilities/Md2Html.cs

```diff
@@ -0,0 +1,161 @@
+﻿using System;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+
+namespace Dynamo.Utilities
+{
+    public class Md2Html
+    {
+        private readonly Process process = new Process();
+        private readonly bool started;
+        /// <summary>
+        /// Constructor
+        /// Start the CLI tool and keep it around
+        /// </summary>
+        public Md2Html()
+        {
+            ProcessStartInfo startInfo = new ProcessStartInfo
+            {
+                CreateNoWindow = true,
+                RedirectStandardOutput = true,
+                RedirectStandardInput = true,
+
+                UseShellExecute = false,
+                Arguments = @"",
+                FileName = GetToolPath()
+            };
+
+            process.StartInfo = startInfo;
+            try
+            {
+                process.Start();
+                AppDomain.CurrentDomain.ProcessExit += new EventHandler(ProcessExit);
+                started = true;
+            }
+            catch(Win32Exception)
+            {
+                // Do nothing
+            }
+        }
+
+        /// <summary>
+        /// Destructor
+        /// Kill the CLI tool, if still running
+        /// </summary>
+        ~Md2Html()
+        {
+            KillProcess();
+        }
+
+        /// <summary>
+        /// Converts a markdown string into Html.
+        /// </summary>
+        /// <param name="writer"></param>
+        /// <param name="mdString"></param>
+        /// <param name="mdPath"></param>
+        public void ParseMd2Html(ref StringWriter writer, string mdString, string mdPath)
```

Comment:
why use a `ref` in these parameters - you're not replacing the writer with another writer are you?

Reply:
I am not. But you are right, this API can be simplified to just return a string.